src/native/libs/System.Globalization.Native/CMakeLists.txt

```diff
@@ -94,7 +89,7 @@ if (LOCAL_BUILD)
     # For minipal files
     include_directories(../../)
     include_directories(${CMAKE_CURRENT_BINARY_DIR})
-elseif (CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
+elseif ((CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI) AND NOT RUNTIME_FLAVOR STREQUAL "CoreCLR")
     add_definitions(-DSTATIC_ICU)
     set(NATIVEGLOBALIZATION_SOURCES ${NATIVEGLOBALIZATION_SOURCES} pal_icushim_static.c)
 else()
```

Comment:
I am assuming this is the fallout from https://github.com/dotnet/runtime/pull/82393, what I don't quite understand is what the difference is between what's upstream and what's downstream. It sounds like https://github.com/dotnet/runtime/pull/82393 would be "just what we want", but clearly it isn't. Is the core difference that we build ICU from source here and upstream gets it as a binary from somewhere?

That aside, if we must have NAOT-LLVM be different, I don't think indirecting it through the runtime flavor is useful (it'll always be set to `CoreCLR` by the script anyway), we should just call this difference out and do something simpler, e. g. comment this whole block out.

Reply:
> upstream gets it as a binary from somewhere?

That looks to be it, mono is getting `libicuuc.a` and `libicui18n.a` ultimately from `$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)` .  I can't see where that comes from, the SDK perhaps.
https://github.com/dotnet/runtime/blob/d7a30ba1f1c9ccce26a6a223898b7b9aba186a3a/src/mono/wasm/wasm.proj#L26

I will try to do the same and drop `-s USE_ICU=1` from the emcc build

Reply:
This isn't quite as straightforward as I thought because of https://github.com/emscripten-core/emscripten/issues/17540#issuecomment-1207463409  .  I was going to link `libicuuc.a` and friends into `libSystem.Globalization.Native.a` but that (linking one static library into another) seems hard with cmake as per that comment.  Mono solves this by linking into an executable (`dotnet.native`) which cmake does do.

Would it be acceptable therefore to do something similar to mono and link `libicuuc.a`, `libicui18n.a` and `libicudata.a` from the `microsoft.netcore.runtime.icu.transport` package at "user" link time.  I.e. bring in via a `<PackageReference Include="Microsoft.NETCore.Runtime.ICU.Transport" PrivateAssets="all" Version="$(MicrosoftNETCoreRuntimeICUTransportVersion)" GeneratePathProperty="true" />` from  a build integration file?

Reply:
The transport packages are meant for transport of bits between builds of different dotnet org repos. They are not meant to be used by "user" build.

You would want to repackage the ICU bits from the transport package into the `Microsoft.DotNet.ILCompiler`.

Reply:
I've not gone the repackaging route for now because I was able to just put a cmake post build event in:
```
if (CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
   add_custom_command(TARGET System.Globalization.Native-Static
       POST_BUILD 
       COMMAND ${CMAKE_AR} qL $<TARGET_FILE:System.Globalization.Native-Static> ${CMAKE_ICU_DIR}/lib/libicuuc.a ${CMAKE_ICU_DIR}/lib/libicui18n.a ${CMAKE_ICU_DIR}/lib/libicudata.a)
endif()
```
which resolves to a `llvm-ar` command appending the contents of these 3 archives to `libSystem.Globalization.Native.a`  This seemed simpler, but for it to work with mono, we would have to see if the mono team are ok with removing 
https://github.com/dotnet/runtimelab/blob/2f8f4a90ca6deb76ab2551ba4aca56713267a5cb/src/mono/wasm/runtime/CMakeLists.txt#L19-L20

And understand why we need the data file `libicudata.a` and mono does not (or maybe they do need it for the Aot version?)